src/main/java/com/mraof/minestuck/entity/underling/UnderlingEntity.java

```diff
@@ -38,15 +38,18 @@
 import net.minecraftforge.common.util.Constants;
 import net.minecraftforge.common.util.FakePlayer;
 import software.bernie.geckolib3.core.IAnimatable;
+import software.bernie.geckolib3.core.manager.AnimationFactory;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.util.*;
 
-public abstract class UnderlingEntity extends AnimatedCreatureEntity implements IMob, IAnimatable
+public abstract class UnderlingEntity extends AttackingAnimatedEntity implements IMob, IAnimatable
 {
 	public static final UUID GRIST_MODIFIER_ID = UUID.fromString("08B6DEFC-E3F4-11EA-87D0-0242AC130003");
 	private static final DataParameter<String> GRIST_TYPE = EntityDataManager.defineId(UnderlingEntity.class, DataSerializers.STRING);
+	
+	private final AnimationFactory factory = new AnimationFactory(this);
 	protected final EntityListFilter attackEntitySelector = new EntityListFilter(new ArrayList<>());    //TODO this filter isn't being saved. F1X PLZ
```

Comment:
I know you did not create this, but do you have any suggestions on how to fix?

Reply:
To fix it would involve creating some serialisation functions on the filter and use them in the entitys serialisation functions.
However, I'd like to have a closer look at the filter and determine if that functionality should simply be removed instead, and remove the serialisation need that way instead.

Reply:
If you do not think it is a major issue I am also fine with leaving it

Reply:
The filter tracks which entities that the underling considers to be enemies. If a non-underling entity attacks the underling, it gets added to the filter, so that the specific underling would keep attacking entities of that type. Because the filter is not serialized, the list of enemies gets cleared on unload-reload. (The player entity type is always added back, so it'll still consider players to be enemies)

This is a minor issue, and not a major issue.